### PR TITLE
feat(cli): add ChainSpecParser type to rethCli

### LINKED ISSUE #11770: Integrate ChainSpec(Parser) into RethCli
Currently we rely on:

https://github.com/paradigmxyz/reth/blob/7b1b1fcb3b632d8c877a3e761a77ae4d513c0c2a/crates/cli/commands/src/node.rs#L24-L25

if we want to integrate more things that should come with a default but we wan to override we need to expand this functionality

for this we should first integrate the chainspec parser into `RethCli` type as associated type.

 https://github.com/paradigmxyz/reth/blob/7b1b1fcb3b632d8c877a3e761a77ae4d513c0c2a/crates/cli/cli/src/lib.rs#L24-L24

### PR DESCRIPTION
**Pull request by aroralanuk**
I've added added  `ChainSpecParser` type to `RethCli` trait and updated the `NodeCommand` struct to use this instead.

Fixes https://github.com/paradigmxyz/reth/issues/11770

### CODE REVIEW

File: crates/cli/commands/src/node.rs
Comment (mattsse) [inline crates/cli/commands/src/node.rs:1]: I want to do this is a separate pr, once we're happy with the rethcli and we've figured out better ergonomics, so please keep this as before for now

File: crates/cli/cli/src/lib.rs [hunk 1]
```diff
@@ -22,6 +23,9 @@ pub mod chainspec;
 /// It provides commonly used functionality for running commands and information about the CL, such
 /// as the name and version.
 pub trait RethCli: Sized {
+    /// The associated ChainSpecParser type
+    type ChainSpecParser: ChainSpecParser;
```
Comment (mattsse) [inline crates/cli/cli/src/lib.rs:27]: this is good!

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/cli/commands/src/node.rs
Comment (aroralanuk) [reply crates/cli/commands/src/node.rs:1]: Done!

**Review by mattsse — APPROVED** [review]

**Comment by aroralanuk** [comment]
Sorry again :|

**Comment by aroralanuk** [comment]
this can also be merged now @mattsse

**Pull request merged**